src/helper.cc

```diff
@@ -1286,13 +1298,10 @@ GetFirstAvailable(const helper * hlp)
     for (n = hlp->servers.head; n != NULL; n = n->next) {
         srv = (helper_server *)n->data;
 
-        if (srv->canAcceptRequests())
-            continue;
-
         if (selected && selected->stats.pending <= srv->stats.pending)
             continue;
 
-        if (srv->flags.shutdown)
+        if (!srv->available())
             continue;
 
         if (!srv->stats.pending)
```

Comment:
Please check the `selected` assignment logic just below this check (I cannot attach a review comment there). It looks like we are not selecting the least loaded server because of that "break" statement -- to find the least loaded available server, we have to look through _all_ available servers and cannot quit early, right? If that code is indeed broken, we should either fix it in this PR or add an XXX.